pkg/collector/reconcile/config_replace.go

```diff
@@ -71,13 +87,16 @@ func ReplaceConfig(instance v1alpha1.OpenTelemetryCollector) (string, error) {
 		}
 	}
 
-	updPromCfgMap := make(map[string]interface{})
-	if err := mapstructure.Decode(cfg, &updPromCfgMap); err != nil {
-		return "", err
+	if EnableTargetAllocatorRewrite.IsEnabled() {
+		cfg.TargetAllocConfig = &targetAllocator{
```

Comment:
General/basic question - Where will this new cfg field end up being used? Is there another file that will consume this `TargetAllocConfig` and do something with it?

Reply:
this is just used as a convenience for serialization, not really meant to be used beyond that. see [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusreceiver#opentelemetry-operator) for more details